samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -324,4 +398,45 @@ public void subscribeToProcessorChange(IZkChildListener listener) {
     zkClient.subscribeChildChanges(keyBuilder.getProcessorsPath(), listener);
     metrics.subscriptions.inc();
   }
+
+  /**
+   * Represents zookeeper processor node.
+   */
+  private static class ProcessorNode {
```

Comment:
Can you explain the motivation for introducing `ProcessorNode` here? Is it possible to achieve the same by adding the ephemeralProcessorPath to `ProcessorData` ?